addon/components/frost-action-bar.js

```diff
@@ -80,25 +93,162 @@ export default Component.extend({
+      controls.reverse()
+    }
+
+    return controls
   },
 
   @readOnly
-  @computed('isControlsObject', 'controls', 'selectedItems.[]')
-  _controls (isControlsObject, controls, selectedItems) {
-    if (isControlsObject) {
-      return applicableControls(this.selectedTypesWithControls(selectedItems)).map((controlName) => {
+  @computed('controls', 'selectedItems.[]', 'moreActions')
+  /**
+   * gets controls for the action bar
+   * @param {Ember.Component[]|object} controls - array or hash of moreActions
+   * @param {Ember.Object[]|object[]} selectedItems - array of selected items for filtering moreActions
+   * @param {Ember.Component[]|boolean|object[]|object} moreActions - array or hash of extra moreActions
+   *  or flag indicating whether to generate them automatically
+   * @returns {object[]} array of moreActions definitions
+   */
+  _moreActions (controls, selectedItems, moreActions) {
+    controls = this.getNormalizedControls(controls, selectedItems)
+    moreActions = this.getNormalizedControls(moreActions, selectedItems)
+
+    // fail fast if we're told not use this feature
+    if (moreActions === false) {
+      return []
+    }
+
+    // get the extra buttons if needed (default)
+    if (moreActions === true) {
+      let buttonCount = controls.filter(control => !this.isFrostButton(control)).length
+
+      // reverse to grab buttons from the end
+      moreActions = controls.reverse()
+        .filter(control => this.isFrostButton(control) && ++buttonCount > MAX_CONTROLS)
+      moreActions.reverse()
+    }
+
+    // convert buttons to POJOs if needed
+    return moreActions.map(button => this.isFrostButton(button) ? this.convertButton(button) : button)
+  },
+
+  // == Functions =============================================================
+
+  /**
+   * converts a button to a hash of arguments to be used for building the moreActions button
+   * @param {FrostButton} button - a frost-button to convert
+   * @returns {object} - plain object of props for moreActions button
+   */
+  convertButton (button) {
+    const propNames = ['disabled', 'hook', 'onClick', 'text']
+
+    // ember 2.12+
+    if (button.args) {
+      return button.args.named.keys.reduce((props, key) => {
+        // only get specified prop names
+        if (propNames.includes(key)) {
+          // compute the disabled property if exists and is needed
+          if (key === 'disabled') {
+            let disabled = get(button, 'args.named._map.disabled')
+            let compute = get(disabled, 'compute')
+
+            // need disabled.compute vs just compute for context
+            props[key] = typeof compute === 'function' ? disabled.compute() : disabled
+
+          // else grab value from component
+          } else {
+            // props[key] = get(button, `args.named._map.${key}.inner`)
+            props[key] = button.args.named._map[key].inner
+          }
+        }
+
+        return props
+      }, {})
+    // ember 2.8
```

Comment:
https://github.com/ciena-frost/ember-frost-list/blob/master/addon/utils/get-component-name.js has code related to this that may provide additional consideration (I have not evaluated it).

Reply:
yeah - I'm essentially doing the same thing - perhaps that should be moved to some shared utils in a core repo?

Reply:
Eventually/likely, yes.  It's on my list of things to look at around creating a repo in the frost ecosystem to contain such things for inclusion.

Reply:
I don't know that we need any more repos... could it be part of frost-core?

Reply:
If every repo we have/use or would need to use this code in already depends on `ember-frost-core` then perhaps but if not then having it separate would be preferred (until tree shaking is complete).